plugin/src/main/java/io/jenkins/plugins/casc/SchemaGeneration.java

```diff
@@ -0,0 +1,197 @@
+
+    public static JSONObject generateSchema() {
+
+        /**
+         * The initial template for the JSON Schema
+         */
+        JSONObject schemaObject = new JSONObject(schemaTemplateObject.toString());
+        /**
+         * This generates the schema for the root configurators
+         * Iterates over the root elements and adds them to the schema.
+         */
+        schemaObject.put("properties", generateRootConfiguratorObject());
+        /**
+         * This generates the schema for the base configurators
+         * Iterates over the base configurators and adds them to the schema.
+         */
+        JSONObject schemaConfiguratorObjects = new JSONObject();
+        ConfigurationAsCode configurationAsCodeObject = ConfigurationAsCode.get();
+        for (Object configuratorObject : configurationAsCodeObject.getConfigurators()) {
+            if (configuratorObject instanceof BaseConfigurator) {
+                BaseConfigurator baseConfigurator = (BaseConfigurator) configuratorObject;
+                List<Attribute> baseConfigAttributeList = baseConfigurator.getAttributes();
+                if (baseConfigAttributeList.size() == 0) {
+                    schemaConfiguratorObjects
+                        .put(((BaseConfigurator) configuratorObject).getTarget().getSimpleName().toLowerCase(),
+                            new JSONObject()
+                                .put("type", "object")
+                                .put("properties", new JSONObject()));
+
+                } else {
+                    JSONObject attributeSchema = new JSONObject();
+                    for (Attribute attribute : baseConfigAttributeList) {
+                        if (attribute.multiple) {
+                            generateMultipleAttributeSchema(attributeSchema, attribute);
+                        } else {
+                            if (attribute.type.isEnum()) {
+                                generateEnumAttributeSchema(attributeSchema, attribute);
+                            } else {
+                                attributeSchema.put(attribute.getName(), generateNonEnumAttributeObject(attribute));
+                                schemaConfiguratorObjects
+                                    .put(((BaseConfigurator) configuratorObject).getTarget().getSimpleName().toLowerCase(),
+                                        new JSONObject()
+                                            .put("type", "object")
+                                            .put("properties", attributeSchema));
+                            }
+                        }
+                    }
+                }
+            }
+            /**
+             * Used to generate the schema for the implementors of
+             * the HetroDescribable Configurator
+             * It mimics the HetroDescribable Configurator.jelly
+             */
+            else if (configuratorObject instanceof HeteroDescribableConfigurator) {
```

Comment:
Brace should be on same line of else if in java code style, 
I.e
} else if

Reply:
Separated it due to readibilty, do you think it should be placed back up ?

Reply:
Please do